bundles/org.openhab.binding.tado/src/main/java/org/openhab/binding/tado/internal/api/TadoApiTypeUtils.java

```diff
@@ -144,6 +150,85 @@ public static AcFanSpeed getAcFanSpeed(FanSpeed fanSpeed) {
         return null;
     }
 
+    public static ACFanLevel getFanLevel(FanLevel fanLevel) {
+        if (fanLevel == null) {
+            return null;
+        }
```

Comment:
Would it be possible to add `@NonNullByDefault` annotation to this class, and avoid accepting and returning null? It doesn't make sense to use this method for that purpose. :)

It seems you currently don't have any code paths that would actually pass null to this method. It would even be better to throw `IllegalArgumentException`. But of course null annotations would be preferable.

Reply:
@jlaur currently none of the existing classes in this binding have proper null annotations or null checks. I am intending to fix all of that in a further PR after this one. I am trying to follow the principle of focussing the scope of each PR on a specific clear topic. => So is it Ok with you if I do NOT address this particular comment (nor the similar ones below) in this current PR, but defer it to the next forthcoming PR?

Reply:
See #12552

Reply:
I agree with this approach, so we can skip null annotation refactoring for now. We just need to make sure that this PR in its own right will work correctly and not introduce possible NullPointerExceptions.

Reply:
>  not introduce possible NullPointerExceptions.

Agreed. See my response concerning `getFanLevel()` below..